### PR TITLE
[ONL-7919] ec-table-head to TS

### PR DESCRIPTION
**Pull request by raulwwq0**

### CODE REVIEW

File: src/components/ec-table-head/ec-table-head.vue [hunk 1]
```diff
@@ -49,47 +49,52 @@
...
+const props = withDefaults(defineProps<TableHeadProps>(), {
+  columns: () => [],
+  sorts: () => [],
 });
 
-function getSortDirection(column) {
+function getSortDirection(column: TableHeadColumn): SortDirection | undefined {
   const existingSort = props.sorts.find(sort => sort.column === column.name);
   return existingSort && existingSort.direction;
 }
```
Comment (wolandec) [inline src/components/ec-table-head/ec-table-head.vue:84]:
```suggestion
function getSortDirection(column: TableHeadColumn) {
  return props.sorts.find(sort => sort.column === column.name)?.direction;
}
```

File: src/components/ec-table-head/ec-table-head.vue [hunk 2]
```diff
@@ -49,47 +49,52 @@
...
 
-function onSort(column) {
+function onSort(column: TableHeadColumn): void {
   emit('sort', column);
 }
```
Comment (wolandec) [inline src/components/ec-table-head/ec-table-head.vue:88]:
```suggestion
function onSort(column: TableHeadColumn) {
  emit('sort', column);
}
```
TS infers such kind of types from the function code

File: src/components/ec-table-head/ec-table-head.vue [hunk 3]
```diff
@@ -49,47 +49,52 @@
...
 
-function getWidthStyle(column) {
+function getWidthStyle(column: TableHeadColumn): StyleValue {
   if (column && (column.maxWidth || column.minWidth)) {
     return { maxWidth: column.maxWidth, minWidth: column.minWidth };
   }
-  return null;
+  return null as unknown as StyleValue;
```
Comment (wolandec) [inline src/components/ec-table-head/ec-table-head.vue:94]:
```suggestion
  return null;
```
null doesn't match type StyleValue

File: src/components/ec-table-head/ec-table-head.vue [hunk 4]
```diff
@@ -49,47 +49,52 @@
...
   return existingSort && existingSort.direction;
 }
 
-function onSort(column) {
+function onSort(column: TableHeadColumn): void {
   emit('sort', column);
 }
 
-function getWidthStyle(column) {
+function getWidthStyle(column: TableHeadColumn): StyleValue {
```
Comment (wolandec) [inline src/components/ec-table-head/ec-table-head.vue:90]:
```suggestion
function getWidthStyle(column: TableHeadColumn) {
```

File: src/components/ec-table-head/ec-table-head.vue [hunk 5]
```diff
@@ -49,47 +49,52 @@
...
   if (column && (column.maxWidth || column.minWidth)) {
     return { maxWidth: column.maxWidth, minWidth: column.minWidth };
   }
-  return null;
+  return null as unknown as StyleValue;
 }
 
-function getStickyColumnClass(colIndex, columns) {
+function getStickyColumnClass(colIndex: number, columns: TableHeadColumn[]): string | null {
```
Comment (wolandec) [inline src/components/ec-table-head/ec-table-head.vue:97]:
```suggestion
function getStickyColumnClass(colIndex: number) {
```

Comment (raulwwq0) [reply src/components/ec-table-head/ec-table-head.vue:97]: We can't remove `columns` because is needed to get `columns.length`

Comment (wolandec) [reply src/components/ec-table-head/ec-table-head.vue:97]: We can use props.columns

File: src/components/ec-table-head/ec-table-head.vue (hunk 3 above)
Comment (wolandec) [reply src/components/ec-table-head/ec-table-head.vue:94]: Where was the problem in template or in this function?

File: src/components/ec-table-head/ec-table-head.vue (hunk 5 above)
Comment (raulwwq0) [reply src/components/ec-table-head/ec-table-head.vue:97]: oh... makes sense... I'm changing it!

File: src/components/ec-table-head/ec-table-head.vue (hunk 3 above)
Comment (raulwwq0) [reply src/components/ec-table-head/ec-table-head.vue:94]: In the template

Comment (wolandec) [reply src/components/ec-table-head/ec-table-head.vue:94]: I think we could return undefined then, instead of null.

**Review by wolandec — APPROVED** [review]

**Review by wolandec — APPROVED** [review]

**Pull request merged**